### PR TITLE
add integration tests to grunt test

### PR DESCRIPTION
**Pull request by trevj**
So, we've had this integration test for the TCP wrapper since around the time we moved to Freedom 0.6. It was a little experiment and wasn't really reviewed (my bad: it got lumped in with some other changes). I'd like to get another pair of eyes on it now, and get it running as part of `grunt test`.

Then we can tackle the end-to-end SOCKS test:
https://github.com/uProxy/uproxy/issues/756

ONE THING: I've noticed this is currently flaky...tracking it down, but this test fails the first time the extension loads. If you reload the extension, it works fine. Just that first time it fails...I think if we could figure out why, it would explain why Simple SOCKS mysteriously fails to start occasionally:
https://github.com/uProxy/uproxy/issues/628

[Review on Reviewable](https://reviewable.io/reviews/uproxy/uproxy-networking/178)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -50,8 +50,7 @@
     "freedom": "^0.6.11"
   },
   "scripts": {
-    "test": "grunt test",
-    "prepublish": "grunt test",
+    "test": "grunt build jasmine",
```
Comment (bemasc) [inline package.json]: What is the motivation for this change?

Comment (iislucas) [inline package.json]: I would expect this line to be 

```
 "test": "grunt test",
```

w.r.t. removal of `prepublish` I think that makes sense: it is not the right place for pre-submit tests and is deprecated.

Comment (trevj) [inline package.json]: So, I changed the test line to prevent Travis running the integration tests (it runs `npm test`). This is not the only way, e.g. I could add a new target like `grunt test_headless` but felt that was overkill.

Comment (bemasc) [inline package.json]: Is that because they're flaky, or because they're slow?

Comment (trevj) [inline package.json]: Neither: Chrome isn't available in the Travis environment so the integration tests can't be run. That's the only reason.

Comment (iislucas) [inline package.json]: OK, that makes sense. How about we make a `unit-tests` target and an `integration-tests` target, and each of those runs the relevant tests, and then it'll be super clear what's happening. :)

**Comment by iislucas** [comment]
:+1:  from me (once you've picked your favorite solution for naming test build targets)

File: package.json (hunk 1 above)
Comment (trevj) [inline package.json]: I factored out `unit_test` and `integration_test` from `grunt test`.

**Comment by trevj** [comment]
Thanks for the review!
